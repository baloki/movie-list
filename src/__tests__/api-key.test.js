import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import ApiKey from '../api-key';

Enzyme.configure({ adapter: new Adapter() });

describe('ApiKey', () => {
  it('fires the parent event when the submit button is clicked', () => {
    const mockCallBack = jest.fn();

    const component = shallow((<ApiKey submitMethod={mockCallBack} />));
    component.find('button').simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });

  it('renders correctly', () => {
    const mockCallBack = jest.fn();

    const tree = renderer
      .create(<ApiKey submitMethod={mockCallBack} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
