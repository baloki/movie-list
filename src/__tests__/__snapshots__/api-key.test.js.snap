// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiKey renders correctly 1`] = `
<div
  className="api-key"
>
  <p>
    Before you can use this application you need to provide an API key from the 
    <a
      href="http://www.omdbapi.com/apikey.aspx"
      rel="noopener noreferrer"
      target="_blank"
    >
      OMDb API
    </a>
     site.  Please sign-up there and obtain a key before proceeding.
  </p>
  <input
    className="api-key__field"
    placeholder="Insert API Key Here"
  />
  <button
    className="api-key__button"
    onClick={[Function]}
    type="submit"
  >
    Submit
  </button>
</div>
`;
